src/components/structures/MatrixChat.js

```diff
@@ -275,11 +279,31 @@ module.exports = React.createClass({
                     screen: 'post_registration'
                 });
                 break;
+            case 'start_login_from_guest':
```

Comment:
yup, it could be combined together.  i guess it was a choice of separating the two flows with an isGuest() or adding a different dispatch target.  keeping it separate gives us the option of logging in without preserving the guest details if desired (e.g. if you explicitly wanted to replace your current session with a new login).  i'm not sure that one is a huge win over the other, but yell if you feel strongly.